deps/rabbit/src/rabbit_stream_coordinator.erl

```diff
@@ -185,8 +185,15 @@ apply(#{from := From}, {policy_changed, #{stream_id := StreamId}} = Cmd,
                 Conf ->
                     %% No changes, ensure we only trigger an election if it's a must
                     {State, ok, []};
-                _ ->
-                    {State, ok, [{mod_call, osiris_writer, stop, [Conf]}]}
+                #{retention := Retention,
+                  leader_pid := Pid} = Conf0 ->
+                    case maps:remove(retention, Conf) == maps:remove(retention, Conf0) of
+                        true ->
+                            %% Only retention policy has changed, it doesn't need a full restart
+                            {State, ok, [{mod_call, osiris, update_retention, [Pid, Retention]}]};
```

Comment:
This needs to be done for each member, also this is a call so should be done in a separate process with failure handling.

Reply:
Done